helm/3.5.2/share/man/man1/helm.1
.nh
.TH "HELM" "1" "Feb 2021" "Auto generated by spf13/cobra" ""

.SH NAME
.PP
helm \- The Helm package manager for Kubernetes.


.SH SYNOPSIS
.PP
\fBhelm [flags]\fP


.SH DESCRIPTION
.PP
The Kubernetes package manager

.PP
Common actions for Helm:

.RS
.IP \(bu 2
helm search:    search for charts
.IP \(bu 2
helm pull:      download a chart to your local directory to view
.IP \(bu 2
helm install:   upload the chart to Kubernetes
.IP \(bu 2
helm list:      list releases of charts

.RE

.PP
Environment variables:

.TS
allbox;
l l 
l l .
\fB\fCName\fR	\fB\fCDescription\fR
$HELM\_CACHE\_HOME	T{
set an alternative location for storing cached files.
T}
$HELM\_CONFIG\_HOME	T{
set an alternative location for storing Helm configuration.
T}
$HELM\_DATA\_HOME	T{
set an alternative location for storing Helm data.
T}
$HELM\_DEBUG	T{
indicate whether or not Helm is running in Debug mode
T}
$HELM\_DRIVER	T{
set the backend storage driver. Values are: configmap, secret, memory, postgres
T}
T{
$HELM\_DRIVER\_SQL\_CONNECTION\_STRING
T}
	T{
set the connection string the SQL storage driver should use.
T}
$HELM\_MAX\_HISTORY	T{
set the maximum number of helm release history.
T}
$HELM\_NAMESPACE	T{
set the namespace used for the helm operations.
T}
$HELM\_NO\_PLUGINS	T{
disable plugins. Set HELM\_NO\_PLUGINS=1 to disable plugins.
T}
$HELM\_PLUGINS	T{
set the path to the plugins directory
T}
$HELM\_REGISTRY\_CONFIG	T{
set the path to the registry config file.
T}
$HELM\_REPOSITORY\_CACHE	T{
set the path to the repository cache directory
T}
$HELM\_REPOSITORY\_CONFIG	T{
set the path to the repositories file.
T}
$KUBECONFIG	T{
set an alternative Kubernetes configuration file (default "\~/.kube/config")
T}
$HELM\_KUBEAPISERVER	T{
set the Kubernetes API Server Endpoint for authentication
T}
$HELM\_KUBECAFILE	T{
set the Kubernetes certificate authority file.
T}
$HELM\_KUBEASGROUPS	T{
set the Groups to use for impersonation using a comma\-separated list.
T}
$HELM\_KUBEASUSER	T{
set the Username to impersonate for the operation.
T}
$HELM\_KUBECONTEXT	T{
set the name of the kubeconfig context.
T}
$HELM\_KUBETOKEN	T{
set the Bearer KubeToken used for authentication.
T}
.TE

.PP
Helm stores cache, configuration, and data based on the following configuration order:

.RS
.IP \(bu 2
If a HELM\_*\_HOME environment variable is set, it will be used
.IP \(bu 2
Otherwise, on systems supporting the XDG base directory specification, the XDG variables will be used
.IP \(bu 2
When no other location is set a default location will be used based on the operating system

.RE

.PP
By default, the default directories depend on the Operating System. The defaults are listed below:

.TS
allbox;
l l l l 
l l l l .
\fB\fCOperating System\fR	\fB\fCCache Path\fR	\fB\fCConfiguration Path\fR	\fB\fCData Path\fR
Linux	$HOME/.cache/helm	$HOME/.config/helm	$HOME/.local/share/helm
macOS	$HOME/Library/Caches/helm	$HOME/Library/Preferences/helm	$HOME/Library/helm
Windows	%TEMP%\\helm	%APPDATA%\\helm	%APPDATA%\\helm
.TE


.SH OPTIONS
.PP
\fB\-\-debug\fP[=false]
	enable verbose output

.PP
\fB\-h\fP, \fB\-\-help\fP[=false]
	help for helm

.PP
\fB\-\-kube\-apiserver\fP=""
	the address and the port for the Kubernetes API server

.PP
\fB\-\-kube\-as\-group\fP=[]
	group to impersonate for the operation, this flag can be repeated to specify multiple groups.

.PP
\fB\-\-kube\-as\-user\fP=""
	username to impersonate for the operation

.PP
\fB\-\-kube\-ca\-file\fP=""
	the certificate authority file for the Kubernetes API server connection

.PP
\fB\-\-kube\-context\fP=""
	name of the kubeconfig context to use

.PP
\fB\-\-kube\-token\fP=""
	bearer token used for authentication

.PP
\fB\-\-kubeconfig\fP=""
	path to the kubeconfig file

.PP
\fB\-n\fP, \fB\-\-namespace\fP=""
	namespace scope for this request

.PP
\fB\-\-registry\-config\fP="/private/tmp/helm\-20210204\-99981\-1f8vc0y/.brew\_home/Library/Preferences/helm/registry.json"
	path to the registry config file

.PP
\fB\-\-repository\-cache\fP="/private/tmp/helm\-20210204\-99981\-1f8vc0y/.brew\_home/Library/Caches/helm/repository"
	path to the file containing cached repository indexes

.PP
\fB\-\-repository\-config\fP="/private/tmp/helm\-20210204\-99981\-1f8vc0y/.brew\_home/Library/Preferences/helm/repositories.yaml"
	path to the file containing repository names and URLs


.SH SEE ALSO
.PP
\fBhelm\-completion(1)\fP, \fBhelm\-create(1)\fP, \fBhelm\-dependency(1)\fP, \fBhelm\-env(1)\fP, \fBhelm\-get(1)\fP, \fBhelm\-history(1)\fP, \fBhelm\-install(1)\fP, \fBhelm\-lint(1)\fP, \fBhelm\-list(1)\fP, \fBhelm\-package(1)\fP, \fBhelm\-plugin(1)\fP, \fBhelm\-pull(1)\fP, \fBhelm\-repo(1)\fP, \fBhelm\-rollback(1)\fP, \fBhelm\-search(1)\fP, \fBhelm\-show(1)\fP, \fBhelm\-status(1)\fP, \fBhelm\-template(1)\fP, \fBhelm\-test(1)\fP, \fBhelm\-uninstall(1)\fP, \fBhelm\-upgrade(1)\fP, \fBhelm\-verify(1)\fP, \fBhelm\-version(1)\fP


.SH HISTORY
.PP
4\-Feb\-2021 Auto generated by spf13/cobra


helm/3.5.2/.brew/helm.rb
class Helm < Formula
  desc "Kubernetes package manager"
  homepage "https://helm.sh/"
  url "https://github.com/helm/helm.git",
      tag:      "v3.5.2",
      revision: "167aac70832d3a384f65f9745335e9fb40169dc2"
  license "Apache-2.0"
  head "https://github.com/helm/helm.git"

  depends_on "go" => :build

  def install
    system "make", "build"
    bin.install "bin/helm"

    mkdir "man1" do
      system bin/"helm", "docs", "--type", "man"
      man1.install Dir["*"]
    end

    output = Utils.safe_popen_read({ "SHELL" => "bash" }, bin/"helm", "completion", "bash")
    (bash_completion/"helm").write output

    output = Utils.safe_popen_read({ "SHELL" => "zsh" }, bin/"helm", "completion", "zsh")
    (zsh_completion/"_helm").write output
  end

  test do
    system "#{bin}/helm", "create", "foo"
    assert File.directory? "#{testpath}/foo/charts"

    version_output = shell_output("#{bin}/helm version 2>&1")
    assert_match "Version:\"v#{version}\"", version_output
    if build.stable?
      assert_match stable.instance_variable_get(:@resource).instance_variable_get(:@specs)[:revision], version_output
    end
  end
end
